File: C#/7. Queue/QueuesUsingStacks.cs
public class MyQueue {

    private Stack<int> Stack;
    private Stack<int> StackToImpelemnt;
    public MyQueue() {
        Stack = new Stack<int>();
        StackToImpelemnt = new Stack<int>();
    }
    
    public void Push(int x) {
        Stack.Push(x);
    }
    
    public int Pop() {
        while(Stack.Count>0){
            StackToImpelemnt.Push(Stack.Pop());
        }
        int poppedOut =  StackToImpelemnt.Pop();
        while(StackToImpelemnt.Count>0){
            Stack.Push(StackToImpelemnt.Pop());
        }
        return poppedOut;
    }
    
    public int Peek() {
       while(Stack.Count>0){
            StackToImpelemnt.Push(Stack.Pop());
        }
        int peek =  StackToImpelemnt.Peek();
        while(StackToImpelemnt.Count>0){
            Stack.Push(StackToImpelemnt.Pop());
        }
        return peek;
    }
    
    public bool Empty() {
        return Stack.Count==0;
    }
}

File: C++/7. Queue/QueuesUsingStacks.cpp
#include <iostream>
#include <cstdlib>
#include <stack>

using namespace std;
class MyQueue {
    
private:
    stack<int> input, output;

public:
    MyQueue() {
        
    }
    
    void push(int x) {
        input.push(x);
    }
    
    int pop() {
        if(!output.empty()) {
            int x = output.top();
            output.pop();
            return x;
        } else {
            while(!input.empty()) {
                output.push(input.top());
                input.pop();
            }
            int x = output.top();
            output.pop();
            return x;
        }
    }
    
    int peek() {
        if(!output.empty()) {
            return output.top();
        } else {
            while(!input.empty()) {
                output.push(input.top());
                input.pop();
            }
            return output.top();
        }
    }
    
    bool empty() {
        return input.empty() && output.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 * 
**/